src/store/modules/user/actions.ts

```diff
@@ -51,13 +51,16 @@ const actions: ActionTree<UserState, RootState> = {
   /**
    * Get User profile
    */
-  async getProfile ( { commit }) {
+  async getProfile ({ dispatch, commit }) {
     const resp = await UserService.getProfile()
     if (resp.status === 200) {
       const localTimeZone = moment.tz.guess();
       if (resp.data.userTimeZone !== localTimeZone) {
         emitter.emit('timeZoneDifferent', { profileTimeZone: resp.data.userTimeZone, localTimeZone});
       }
+
+      dispatch('getEComStores', { facilityId: resp.data.facilities[0].facilityId })
```

Comment:
Store resp.data.facilities.length > 0 ? resp.data.facilities[0] : {} in a variable and pass to action as well as mutation
```suggestion
      dispatch('getEComStores', { facilityId: resp.data.facilities.length > 0 ? resp.data.facilities[0] : {} })
```